util/pkg/vfs/swiftfs.go

```diff
@@ -325,33 +326,49 @@ func (p *SwiftPath) createBucket() error {
 	}
 }
 
-// ReadFile implements Path::ReadFile.
+// ReadFile implements Path::ReadFile
 func (p *SwiftPath) ReadFile() ([]byte, error) {
-	var ret []byte
-	done, err := RetryWithBackoff(swiftReadBackoff, func() (bool, error) {
-		glog.V(4).Infof("Reading file %q", p)
-
-		opt := swiftobject.DownloadOpts{}
-		result := swiftobject.Download(p.client, p.bucket, p.key, opt)
-		data, err := (&result).ExtractContent()
-		if err == nil {
-			ret = data
-			return true, nil
-		} else if isSwiftNotFound(err) {
-			return true, os.ErrNotExist
-		} else {
-			return false, fmt.Errorf("error reading %s: %v", p, err)
+	var b bytes.Buffer
+	done, err := RetryWithBackoff(gcsReadBackoff, func() (bool, error) {
+		b.Reset()
+		_, err := p.WriteTo(&b)
+		if err != nil {
+			if os.IsNotExist(err) {
+				// Not recoverable
+				return true, err
+			}
+			return false, err
 		}
+		// Success!
+		return true, nil
 	})
 	if err != nil {
 		return nil, err
 	} else if done {
-		return ret, nil
+		return b.Bytes(), nil
 	} else {
+		// Shouldn't happen - we always return a non-nil error with false
 		return nil, wait.ErrWaitTimeout
 	}
 }
 
+// WriteTo implements io.WriterTo
+func (p *SwiftPath) WriteTo(out io.Writer) (int64, error) {
+	glog.V(4).Infof("Reading file %q", p)
+
+	opt := swiftobject.DownloadOpts{}
+	result := swiftobject.Download(p.client, p.bucket, p.key, opt)
+	if result.Err != nil {
+		if isSwiftNotFound(result.Err) {
+			return 0, os.ErrNotExist
```

Comment:
Do we want to fmt this?

Reply:
A lot of callers expect os.ErrNotExist when the file is not found.  I added a comment to `ReadFile`:

```
	// ReadFile returns the contents of the file, or an error if the file could not be read.
	// If the file did not exist, err = os.ErrNotExist
	// As this reads the entire file into memory, consider using WriteTo for bigger files
	ReadFile() ([]byte, error)
```